pkg/costmodel/costmodel.go

```diff
@@ -1075,27 +1075,38 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 		// not all providers are guaranteed to use this, so don't overwrite a Provider assignment if we can't find something under that capacity exists
 		gpuc := 0.0
 		q, ok := n.Status.Capacity["nvidia.com/gpu"]
+		gpuCount := q.Value()
 		_, hasReplicas := n.Labels["nvidia.com/gpu.replicas"]
 
-		if ok && !hasReplicas {
-			gpuCount := q.Value()
-			if gpuCount != 0 {
-				newCnode.GPU = fmt.Sprintf("%d", gpuCount)
-				newCnode.VGPU = newCnode.GPU
-				gpuc = float64(gpuCount)
-			}
-		} else if hasReplicas { // See https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html
-			if q.Value() == 0 {
-				q = n.Status.Capacity["nvidia.com/gpu.shared"]
-			}
+		if ok && gpuCount != 0 && !hasReplicas {
```

Comment:
@thomasvn is there any way I can convince you to break out all this GPU logic into another function? Then you could write a quick unit test

Reply:
this function is uber large as is and could use the love

Reply:
@ameijer Yep good callout! Let's use this as an opportunity to break it out into another function and unit test.

Reply:
Committed my refactor + unit tests. Marking as resolved for now.